### ISSUE TITLE
resolvers don't work if you define them in a separate const

### ISSUE THREAD

**Issue by biltongza** [angular/angular#33404]
# 🐞 bug report

### Is this a regression?
Not sure.

### Description

In our project we have some routes that share some common resolvers. We'd like to not repeat ourselves and so we made a `const resolvers = {}` for these routes. This works locally on our dev machines with `ng serve`, but once built and published to our dev server we noticed that the resolvers we put in this const were no longer working.

## 🔬 Minimal Reproduction
```
git clone https://github.com/biltongza/angular-const-resolvers
cd angular-const-resolvers
ng build -c production
```
Host the resulting build on any web server and navigate to it. Click on Route 1 and Route 2 and observe that the resolvers from `const resolvers` do not work, while the one defined directly in the route config does work. Click on Route 3 to see that all resolvers work if defined directly in the route config.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
❯ ng version

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 8.3.14
Node: 12.13.0
OS: linux x64
Angular: 8.2.12
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.14
@angular-devkit/build-angular     0.803.14
@angular-devkit/build-optimizer   0.803.14
@angular-devkit/build-webpack     0.803.14
@angular-devkit/core              8.3.14
@angular-devkit/schematics        8.3.14
@angular/cli                      8.3.14
@ngtools/webpack                  8.3.14
@schematics/angular               8.3.14
@schematics/update                0.803.14
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.39.2
</code></pre>

**Anything else relevant?**
The issue does NOT appear if you `ng serve`. We use AOT compilation (which is enabled in the repro).

**Comment by trotyl** [comment]
Duplicate of https://github.com/angular/angular/issues/25317

---

It's known issue to ViewEngine, where metadata lowering only supported in specific fields:

https://github.com/angular/angular/blob/4aa51b751bd6b59571afcdd5add3ca0ba4f4b37a/packages/compiler-cli/src/transformers/program.ts#L43

Non-lowering fields will be fully statically resolved, but spread syntax (**it's not an operator**) was not supported, which will be silently ignored without producing error.

This should have been fixed in Ivy, could you verify it at `@angular/*@next` version? (9.0.0-rc.0 at this moment)

**Comment by biltongza** [comment]
Our project currently doesn't compile in 9, will make effort to verify when we get it compiling.

**Comment by atscott** [comment]
Thanks for reporting this issue. Closing as duplicate of #25317 and because it should be fixed in the 9.0 release. If you still encounter this issue once you get your project compiling with v9 [open a new issue](https://github.com/angular/angular/issues/new/choose) and follow the instructions in the issue template.
